addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,517 @@
+            # or it was submitted and rejected. Either way, we need to assign it a new Chain Index and regenerate
+            # the data that depends on it before submitting (UUID, XML content, signature)
+            invoice.l10n_sa_chain_index = invoice.journal_id._l10n_sa_edi_get_next_chain_index()
+            invoice._l10n_sa_generate_unsigned_data()
+
+        # Generate Invoice name for attachment
+        attachment_name = self._l10n_sa_get_invoice_name(invoice)
+
+        # Generate XML, sign it, then submit it to ZATCA
+        response_data, submitted_xml = self._l10n_sa_export_zatca_invoice(invoice)
+
+        # Check for submission errors
+        if response_data.get('error'):
+
+            # If the request was rejected, we save the signed xml content as an attachment
+            if response_data.get('rejected'):
+                invoice._l10n_sa_save_submission(submitted_xml, attachment_name, response_data)
+
+            # If the request returned an exception (Timeout, ValueError... etc.) it means we're not sure if the
+            # invoice was successfully cleared/reported, and thus we keep the Index Chain.
+            # Else, we recalculate the submission Index (ICV), UUID, XML content and Signature
+            if not response_data.get('excepted'):
+                invoice.l10n_sa_chain_index = False
+
+            return response_data
+
+        # Once submission is done with no errors, check submission status
+        cleared_xml = self._l10n_sa_postprocess_einvoice_submission(invoice, submitted_xml, response_data)
+
+        # Save the submitted/returned invoice XML content once the submission has been completed successfully
+        return {
+            'success': True,
+            'attachment': invoice._l10n_sa_save_submission(cleared_xml.encode(), attachment_name)
+        }
+
+    # ====== EDI Format Overrides =======
+
+    def _is_required_for_invoice(self, invoice):
+        """
+            Override to add ZATCA edi checks on required invoices
+        """
+        self.ensure_one()
+        if self.code != 'sa_zatca':
+            return super()._is_required_for_invoice(invoice)
+
+        return self._l10n_sa_edi_is_required_for_invoice(invoice)
+
+    def _check_move_configuration(self, invoice):
+        """
+            Override to add ZATCA compliance checks on the Invoice
+        """
+
+        def _set_missing_partner_fields(missing_fields, name):
+            return _("- Please, set the following fields on the %s: %s") % (name, ', '.join(missing_fields))
+
+        journal = invoice.journal_id
+        company = invoice.company_id
+
+        errors = super()._check_move_configuration(invoice)
+        if self.code != 'sa_zatca' or company.country_id.code != 'SA':
+            return errors
+
+        if invoice.commercial_partner_id == invoice.company_id.partner_id.commercial_partner_id:
+            errors.append(_("- You cannot post invoices where the Seller is the Buyer"))
```

Comment:
"-".join would be more appropriate after the whole thing (hopefully that works well in Arabic as well)